### PR TITLE
Add test case for internal/errors/net.go

### PR DESCRIPTION
**Pull request by hlts2**
Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

### Description:

I added the test cases for internal/errors/net.go and comment.
And fixed code comments more clearly.

**grammar check passed**

### Environment:

- Go Version: 1.15.7
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.13.1

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/errors/net.go [hunk 1]
```diff
@@ -22,16 +22,17 @@ import "time"
 var (
 	// tcp.
 
-	// ErrFailedInitDialer defines the init dialer error.
+	// ErrFailedInitDialer represents an error that the initialization of the dialer failed.
 	ErrFailedInitDialer = New("failed to init dialer")
-	// ErrInvalidDNSConfig defines the invalid DNS config error.
+
+	// ErrInvalidDNSConfig represents a function to generate an error that the configuration of DNS is invalid.
```
Comment (hlts2) [inline internal/errors/net.go:28]:
```suggestion
	// ErrInvalidDNSConfig represents a function to generate an error that the configuration of the DNS is invalid.
```

File: internal/errors/net_test.go [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+			name: "return an ErrInvalidDNSConfig when dnsRefreshDur and dnsCacheExp are the maximum number of int64",
+			args: args{
+				dnsRefreshDur: time.Duration(math.MaxInt64),
+				dnsCacheExp:   time.Duration(math.MaxInt64),
+			},
+			want: want{
+				want: Errorf("dnsRefreshDuration  > dnsCacheExp, %s, %s", time.Duration(math.MaxInt64), time.Duration(math.MaxInt64)),
+			},
+		},
+	}
```
Comment (vankichi) [inline internal/errors/net_test.go:118]: - Could you add below case?
    1. when dnsRefreshDur is nil.
    2. when dnsCacheExp is nil.
    3. when dnsRefreshDur and dnsCacheExp is nil

Comment (hlts2) [reply internal/errors/net_test.go:118]: @vankichi thank you for your comment.
`time.Duration` is a primitive type, so I think we can not add a test case of nil pattern. https://play.golang.org/p/xdI5op3KTUu

From the above, instead of the nil case pattern, I added zero value pattern 🤔 
what do you think? 
if I made a mistake, I am sorry.

Comment (vankichi) [reply internal/errors/net_test.go:118]: thanks.
So I think we have to add below patterns
1. when dnsRefreshDur is 0.
2. when dnsCacheExp is 0.

and you already add the test when both of these value is 0, it'd be better renaming `return an ErrInvalidDNSConfig when dnsRefreshDur is default value(or some  better word), dnsCacheExp is default value (or some more better word) ` from `return an ErrInvalidDNSConfig when dnsRefreshDur is 0, dnsCacheExp is 0`.
also, it'd be better clearly if you remove these values definition from `args{}`.

File: internal/errors/net_test.go [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+			name: "return an ErrInvalidDNSConfig when dnsRefreshDur is empty and dnsCacheExp is 4 minute",
+			args: args{
+				dnsCacheExp: 4 * time.Minute,
+			},
+			want: want{
+				want: New("dnsRefreshDuration  > dnsCacheExp, 0s, 4m0s"),
+			},
+		},
+		{
+			name: "return an ErrInvalidDNSConfig when dnsRefreshDur is 5m0s and dnsCacheExp is empty",
```
Comment (vankichi) [inline internal/errors/net_test.go:105]:
```suggestion
			name: "return an ErrInvalidDNSConfig when dnsRefreshDur is 5 minute and dnsCacheExp is empty",
```

**Review by vankichi — DISMISSED** [review]
LGTM

**Review by kevindiu — DISMISSED** [review]
LGTM
